swagger.yaml

```diff
@@ -0,0 +1,334 @@
+      description: |
+        Return a list of all dimensions which are going to be used to filter on
+      parameters:
+      - $ref: '#/parameters/filter_id'
+      responses:
+        200:
+          description: "A list of dimension URLs"
+          schema:
+            type: array
+            items:
+              type: string
+        404:
+          $ref: '#/responses/FilterJobNotFound'
+        500:
+          $ref: '#/responses/InternalError'
+  /filters/{filter_id}/dimensions/{dimension_name}:
+    parameters:
+      - $ref: '#/parameters/filter_id'
+      - $ref: '#/parameters/dimension_name'
+    post:
+      tags:
+      - "Filter"
+      summary: "Add a dimension to filter on with a list of values"
+      description: "The dimension can only be added into the job if the state is still set to `created` otherwise 400 status code is returned"
+      produces:
+      - "application/json"
+      parameters:
+      - $ref: '#/parameters/dimension'
+      responses:
+        200:
+          description: "The dimension was created"
+        400:
+          description: "The filter job has been locked as it has been `submitted` to be processed"
+        404:
+          $ref: '#/responses/FilterOrDimensionNotFound'
+        500:
+          $ref: '#/responses/InternalError'
+    delete:
+      tags:
+      - "Filter"
+      summary: "Remove a dimension and any values set within the dimension"
+      description: "Remove a dimension and any values set within the dimension"
+      responses:
+        200:
+          description: "The dimension was removed"
+        400:
+          description: "The filter job has been locked as it has been `submitted` to be processed"
+        404:
+          $ref: '#/responses/FilterOrDimensionNotFound'
+        500:
+          $ref: '#/responses/InternalError'
+    get:
+      tags:
+      - "Filter"
+      summary: "Get all values from a dimension which have been set"
+      description: "Get a list of all values which will be used to filter the dimension"
+      responses:
+        200:
+          description: "A list of all values for a dimension was returned"
+          schema:
+            $ref: '#/definitions/Dimension'
+        400:
+          description: "The filter job has been locked as it has been `submitted` to be processed"
+        404:
+          $ref: '#/responses/FilterOrDimensionNotFound'
+        500:
+          $ref: '#/responses/InternalError'
+  /filters/{filter_id}/dimensions/{dimension_name}/options/{value}:
+    parameters:
+      - $ref: '#/parameters/filter_id'
+      - $ref: '#/parameters/dimension_name'
+      - $ref: '#/parameters/value'
+    get:
+      tags:
+      - "Filter"
+      summary: "Check if a value exists within a dimension"
+      description: "Check if a value exists within a dimension"
+      responses:
+        200:
+          description: "Value exists within the dimension"
+        400:
+          $ref: '#/responses/FilterOrDimensionNotFound'
```

Comment:
The reason for a 400 (bad request) here would be because the get request filter_id or dimension_name don't exist.

A 404 (not found) applies when the filter_id and dimension_name exist but the value does not exist.